### PR TITLE
Upgrade corresponding library to make sure it is compatible with php 8.

### PR DESCRIPTION
**Pull request by cobenash**
## Update
1. Add symfony/finder to replace the unmaintained library finder-facade.
2. Change the outdated forked repository `smmccabe/PHP_CodeSniffer` to `cobenash/PHP_CodeSniffer` for testing purpose. It could be changed back if we update the original.

## Pre-release
https://github.com/cobenash/phpdebt/releases

Please take a look and see if it is workable.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,18 +12,15 @@
...
-            "url": "https://github.com/smmccabe/PHP_CodeSniffer.git"
+            "url": "https://github.com/cobenash/PHP_CodeSniffer.git"
         }
     ],
     "require": {
         "squizlabs/php_codesniffer": "dev-api-friendly as 3.x-dev",
-        "drupal/coder": "^8.3",
-        "phpmd/phpmd": "dev-api-friendly",
-        "phploc/phploc": "^6.0"
+        "drupal/coder": "8.3.13",
```
Comment (skyred) [inline composer.json:20]: Don't hard code a dependency version, unless there is a good reason, isn't there?

File: composer.json [hunk 2]
```diff
@@ -12,18 +12,15 @@
     "repositories": [
         {
             "type": "vcs",
-            "url": "https://github.com/smmccabe/phpmd.git"
-        },
-        {
-            "type": "vcs",
-            "url": "https://github.com/smmccabe/PHP_CodeSniffer.git"
+            "url": "https://github.com/cobenash/PHP_CodeSniffer.git"
```
Comment (skyred) [inline composer.json:15]: Let's not start a new fork. Ask @smmccabe to sync changes from upstream

**Review by skyred — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (cobenash) [reply composer.json:20]: @skyred Hey, here is the reason. https://www.drupal.org/project/coder/issues/3262291

I need to lock the version to 8.3.13 for now.

**Comment by smmccabe** [comment]
Merged, although in the future don't combine code standard fixes and version changes in the same patch

File: composer.json (hunk 2 above)
Comment (smmccabe) [reply composer.json:15]: corrected in master after merge

**Pull request merged**